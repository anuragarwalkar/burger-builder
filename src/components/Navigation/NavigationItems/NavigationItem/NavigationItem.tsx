import React from 'react';
import classes from './navigationItem.module.css';
import { NavLink } from 'react-router-dom';

const NavigationItem = ({active, children, link}: any) => (
    <li className={classes.NavigationItem}>
        <NavLink to={link} className={active ? classes.active : ''}>{children}</NavLink>
    </li>
);

export default NavigationItem;