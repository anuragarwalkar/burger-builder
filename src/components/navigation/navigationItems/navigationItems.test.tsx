import React from "react";
import { configure, shallow, ShallowWrapper } from "enzyme";
import NavigationItems from "./navigationItems";
import EnzymeAdapter from "enzyme-adapter-react-16";
import NavigationItem from "./navigationItem/navigationItem";

configure({ adapter: new EnzymeAdapter() });

describe("<NavigationItems />", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<NavigationItems isAuthenticated={false} />);
  });

  it("should render <NavigationItems /> Component", () => {
    const wrapper: ShallowWrapper = shallow(<NavigationItems isAuthenticated={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("It should render two <NavigationItem /> if user is not authenticated", () => {
    wrapper.setProps({ isAuthenticated: false });
    expect(wrapper.find(NavigationItem)).toHaveLength(2);
  });

  it("Should render three <NavigationItem /> if user is authenticated", () => {
    wrapper.setProps({ isAuthenticated: true });
    expect(wrapper.find(NavigationItem)).toHaveLength(3);
  });

  it("Should render have logout link in <NavigationItem /> if user is authenticated", () => {
    wrapper.setProps({ isAuthenticated: true });
    expect(
      wrapper.contains(<NavigationItem link="/logout">Log Out</NavigationItem>)
    ).toEqual(true);
  });
});
