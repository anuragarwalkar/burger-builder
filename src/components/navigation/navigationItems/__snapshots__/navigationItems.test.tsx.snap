// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems /> should render <NavigationItems /> Component 1`] = `ShallowWrapper {}`;
